MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2005,48 +2005,46 @@ private static int ScaleBitrate(int bitrate, string inputVideoCodec, string outp
             return Convert.ToInt32(scaleFactor * bitrate);
         }
 
-        public int? GetAudioBitrateParam(BaseEncodingJobOptions request, MediaStream audioStream)
+        public int? GetAudioBitrateParam(BaseEncodingJobOptions request, MediaStream audioStream, int? outputAudioChannels)
         {
-            return GetAudioBitrateParam(request.AudioBitRate, request.AudioCodec, audioStream);
+            return GetAudioBitrateParam(request.AudioBitRate, request.AudioCodec, audioStream, outputAudioChannels);
         }
 
-        public int? GetAudioBitrateParam(int? audioBitRate, string audioCodec, MediaStream audioStream)
+        public int? GetAudioBitrateParam(int? audioBitRate, string audioCodec, MediaStream audioStream, int? outputAudioChannels)
         {
             if (audioStream == null)
             {
                 return null;
             }
 
-            if (audioBitRate.HasValue && string.IsNullOrEmpty(audioCodec))
+            var inputChannels = audioStream.Channels ?? 0;
+            var outputChannels = outputAudioChannels ?? 0;
+
+            if (audioBitRate.HasValue && (string.IsNullOrEmpty(audioCodec)
+                || string.Equals(audioCodec, "aac", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(audioCodec, "mp3", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(audioCodec, "ac3", StringComparison.OrdinalIgnoreCase)
+                || string.Equals(audioCodec, "eac3", StringComparison.OrdinalIgnoreCase)))
             {
-                return Math.Min(384000, audioBitRate.Value);
+                return (inputChannels, outputChannels) switch
+                {
+                    ( >= 6, >= 6 or 0) => Math.Min(640000, audioBitRate.Value),
+                    ( > 0, > 0) => Math.Min(outputChannels * 128000, audioBitRate.Value),
+                    ( > 0, _) => Math.Min(inputChannels * 128000, audioBitRate.Value),
```

Comment:
It could be configurable, but that's beyond the scope of this pull request. Previously, when transcoding high bit rate surround sound to stereo, it was transcoded to 384 kbps. That's 192kbps per channel, which is usually more that the source source material uses. This pull request caps it at 128kbps per channel, which is still generous with the codecs we use.

Reply:
Your pull request is creating a regression by reducing the audio per channel to 128 kbps per channel.

Reply:
Yeah, so? Do you feel like transcoding 384kbps 6-channel audio track to 384kbps stereo is a worthy use of bandwidth?

Reply:
It should have a configuration option on the server as a start rather than you just changing the hard coded value.  Perhaps someone needs to save even more bandwidth than your change or wants Higher quality audio.   Future improvement would be an option for local lan vs external connections which have different bandwidths. 
 And then the clients should have the ability to request different bandwidths to reduce if server setting does not meet their needs.